test-user-config.el

```diff
@@ -0,0 +1,93 @@
+;;; test-user-config.el --- Description -*- lexical-binding: t; -*-
+
+;;; Example functions
+(cl-defun my-cowsay (&key term)
+  (let* ((bubble-top (concat " " (make-string (+ (length term) 2) ?_) "\n"))
+         (bubble-middle (format "< %s >\n" term))
+         (bubble-bottom (concat " " (make-string (+ (length term) 2) ?-) "\n"))
+         (cow "        \\   ^__^\n         \\  (oo)\\_______\n            (__)\\       )\\/\\\n                ||----w |\n                ||     ||"))
+    (format "%s%s%s%s" bubble-top bubble-middle bubble-bottom cow)))
+
+(cl-defun my-create-file (&key filename contents)
+  "Create a file with FILENAME and CONTENTS, and open it in a new buffer."
+  (interactive "sEnter filename: \nMEnter contents: ")
+  (if (not (file-exists-p filename))
+      (with-temp-buffer
+        (insert contents)
+        (write-file filename)))
+  (split-window)
+  (other-window 1)
+  (find-file filename))
+
+;;; Callable function schema
+(setq! gptel-callable-functions
+       ;; Hard coded variable specifying callable functions. This could be defined in a user's configuration
```

Comment:
Given a list of emacs functions, we should be able to introspect them to generate this schema data if that's preferable.